Source/VideoProgressView.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  VideoProgressView.swift
+//  Demo
+//
+//  Created by Marijn Schilling on 28/06/16.
+//
+//
+
+import UIKit
+
+class VideoProgressView: UIView {
+
+    static let Height = CGFloat(55.0)
+    private static let ProgressBarXMargin = CGFloat(65.0)
+    private static let ProgressBarYMargin = CGFloat(23.0)
+    private static let ProgressBarHeight = CGFloat(6.0)
+
+    private static let TextLabelWidth = CGFloat(36.0)
+    private static let TextLabelHeight = CGFloat(18.0)
+    private static let TextLabelMargin = CGFloat(18.0)
+
+    var duration = 0.0 {
+        didSet {
+            if self.duration != oldValue {
+              self.durationTimeLabel.text = self.timeStringForSeconds(self.duration)
+            }
+        }
+    }
+    var currentTime = 0.0 {
+        didSet {
+            self.currentTimeLabel.text = self.timeStringForSeconds(self.currentTime)
+            UIView.animateWithDuration(0.2) {
+                self.setFrameForProgressBar()
+            }
+        }
+    }
+
+    var progressPercentage : Double {
+        guard self.currentTime != 0.0 && self.duration != 0.0 else {
+            return 0.0
+        }
+
+        return currentTime/duration
+    }
+
+    var widthForBar : CGFloat {
+        return self.bounds.width - (2 * VideoProgressView.ProgressBarXMargin)
+    }
+
+    var widthForProgressBar : CGFloat {
+        return widthForBar * CGFloat(progressPercentage)
+    }
+
+    lazy var maskBarForRoundedCorners : UIView = {
+        let maskView = UIView()
+        maskView.backgroundColor = UIColor.clearColor()
+        maskView.layer.cornerRadius = ProgressBarHeight/2
+        maskView.clipsToBounds = true
+        maskView.layer.masksToBounds = true;
+        return maskView
+    }()
+
+
+    lazy var backgroundBar : UIView = {
+        let backgroundBar = UIView()
+        backgroundBar.backgroundColor = UIColor.whiteColor()
+        backgroundBar.alpha = 0.2
+        return backgroundBar
+    }()
+
+    lazy var progressBar : UIView = {
+        let progressBar = UIView()
```

Comment:
Great usage of lazy loading 👏 

Only one comment, the `:` should go next to the variable.

```
lazy var progressBar: UIView = {
    //....
}()
```